src/auth/models.cpp

```diff
@@ -101,6 +100,24 @@ std::string PermissionLevelToString(PermissionLevel level) {
   }
 }
 
+FineGrainedAccessPermissions Merge(const FineGrainedAccessPermissions &first,
+                                   const FineGrainedAccessPermissions &second) {
+  std::unordered_map<std::string, uint64_t> permissions{first.permissions()};
+  std::optional<uint64_t> global_permission = std::nullopt;
+
+  if (second.global_permission().has_value()) {
+    global_permission = second.global_permission().value();
+  } else if (first.global_permission().has_value()) {
+    global_permission = first.global_permission().value();
+  }
```

Comment:
Hmm I don't quite understand this logic, what if first and second have values you would just ignore values in first `FineGrainedAccessPermissions`, and why?

Reply:
Closing this comment, was resolved in private discussion between me and Boris